`sv2_ffi` lib crate publishing is broken

this issue is somewhat related to #808 , but I'm submitting it as a separate issue because the solution will not be straightforward like #809 

[`sv2_ffi`publishing is broken with the following error message](https://github.com/stratum-mining/stratum/actions/runs/8383419811/job/22959094437#step:20:9):

```
error[E0277]: the trait bound `Sv2Message<'static>: binary_codec_sv2::codec::Fixed` is not satisfied
   --> src/lib.rs:443:27
    |
443 | pub struct DecoderWrapper(StandardDecoder<Sv2Message<'static>>);
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `binary_codec_sv2::codec::Fixed` is not implemented for `Sv2Message<'static>`, which is required by `Sv2Message<'static>: binary_codec_sv2::codec::GetSize`
    |
    = help: the following other types implement trait `binary_codec_sv2::codec::Fixed`:
              bool
              u8
              u16
              u32
              u64
              f32
              binary_codec_sv2::datatypes::copy_data_types::U24
    = note: required for `Sv2Message<'static>` to implement `binary_codec_sv2::codec::GetSize`
note: required by a bound in `codec_sv2::decoder::WithoutNoise`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/codec_sv2-1.0.0/src/decoder.rs:186:53
    |
186 | pub struct WithoutNoise<B: IsBuffer, T: Serialize + binary_sv2::GetSize> {
    |                                                     ^^^^^^^^^^^^^^^^^^^ required by this bound in `WithoutNoise`

error[E02[77](https://github.com/stratum-mining/stratum/actions/runs/8383419811/job/22959094437#step:20:78)]: the trait bound `binary_codec_sv2::codec::encodable::EncodableField<'_>: From<Sv2Message<'static>>` is not satisfied
   --> src/lib.rs:443:27
    |
443 | pub struct DecoderWrapper(StandardDecoder<Sv2Message<'static>>);
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<Sv2Message<'static>>` is not implemented for `binary_codec_sv2::codec::encodable::EncodableField<'_>`, which is required by `Sv2Message<'static>: binary_codec_sv2::codec::encodable::Encodable`
    |
    = help: the following other types implement trait `From<T>`:
              <binary_codec_sv2::codec::encodable::EncodableField<'decoder> as From<Header>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<bool>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<u8>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<u16>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<binary_codec_sv2::datatypes::copy_data_types::U24>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<u32>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<f32>>
              <binary_codec_sv2::codec::encodable::EncodableField<'a> as From<u64>>
            and 45 others
    = note: required for `Sv2Message<'static>` to implement `Into<binary_codec_sv2::codec::encodable::EncodableField<'_>>`
    = note: required for `Sv2Message<'static>` to implement `binary_codec_sv2::codec::encodable::Encodable`
note: required by a bound in `codec_sv2::decoder::WithoutNoise`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/codec_sv2-1.0.0/src/decoder.rs:1[86](https://github.com/stratum-mining/stratum/actions/runs/8383419811/job/22959094437#step:20:87):41
    |
186 | pub struct WithoutNoise<B: IsBuffer, T: Serialize + binary_sv2::GetSize> {
    |                                         ^^^^^^^^^ required by this bound in `WithoutNoise`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `sv2_ffi` (lib) due to 2 previous errors
error: failed to verify package tarball
Error: Process completed with exit code [101](https://github.com/stratum-mining/stratum/actions/runs/8383419811/job/22959094437#step:20:102).
```

--- a/.github/workflows/release-libs.yaml
+++ b/.github/workflows/release-libs.yaml
@@ -99,7 +99,7 @@ jobs:
         continue-on-error: true
         run: |
           cd protocols/v2/sv2-ffi
-          cargo publish
+          cargo publish --all-features
       - name: Publish crate roles_logic_sv2
         continue-on-error: true
         run: |

--- a/roles/pool/Cargo.toml
+++ b/roles/pool/Cargo.toml
@@ -18,7 +18,7 @@ buffer_sv2 = { version = "^1.0.0", path = "../../utils/buffer" }
 codec_sv2 = { version = "1.0.0", path = "../../protocols/v2/codec-sv2", features = ["noise_sv2"] }
 const_sv2 = { version = "^1.0.0", path = "../../protocols/v2/const-sv2" }
 network_helpers_sv2 = { version = "1.0.0", path = "../roles-utils/network-helpers", features =["with_tokio","with_buffer_pool"] }
-noise_sv2 = { version = "1.0.0", path = "../../protocols/v2/noise-sv2" }
+noise_sv2 = { version = "1.1.0", path = "../../protocols/v2/noise-sv2" }
 rand = "0.8.4"
 roles_logic_sv2 = { version = "^1.0.0", path = "../../protocols/v2/roles-logic-sv2" }
 serde = { version = "1.0.89", features = ["derive", "alloc"], default-features = false }

--- a/roles/roles-utils/rpc/Cargo.toml
+++ b/roles/roles-utils/rpc/Cargo.toml
@@ -6,7 +6,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-stratum-common = { path = "../../../common" }
+stratum-common = { version = "1.0.0", path = "../../../common" }
 serde = { version = "1.0.89", features = ["derive", "alloc"], default-features = false }
 serde_json = { version = "1.0", default-features = false, features = ["alloc","raw_value"] }
 hex = "0.4.3"

